widgets/heatmap/heatmap.go

```diff
@@ -89,42 +144,161 @@ func (hp *HeatMap) ClearYLabels() {
 	hp.yLabels = nil
 }
 
-// ValueCapacity returns the number of values that can fit into the canvas.
+// ValueCapacity returns the number of rows and columns of the heat map
+// that can fit into the canvas.
+//
 // This is essentially the number of available cells on the canvas as observed
 // on the last call to draw. Returns zero if draw wasn't called.
 //
 // Note that this capacity changes each time the terminal resizes, so there is
 // no guarantee this remains the same next time Draw is called.
 // Should be used as a hint only.
-func (hp *HeatMap) ValueCapacity() int {
-	return 0
+func (hp *HeatMap) ValueCapacity() (rows, cols int) {
+	hp.mu.Lock()
+	defer hp.mu.Unlock()
+
+	if hp.lastWidth == 0 || hp.lastHeight == 0 {
+		return 0, 0
+	}
+
+	rows = hp.lastHeight - 1
+	var cw int
```

Comment:
The remaining section of this function could use some unwrapping to make it easier to read. We could start by giving `cw` a better name. Next we could unwrap the `cols` calculation into multiple statements that will either become self-explanatory, or we can add inline comments.

As it stands, I am having a hard time understanding the calculation. Can you help out?